sdk/core/azure-core/src/main/java/com/azure/core/util/IterableStream.java

```diff
@@ -28,15 +32,30 @@
  * @see Iterable
  */
 public class IterableStream<T> implements Iterable<T> {
+    private final ClientLogger logger = new ClientLogger(IterableStream.class);
     private final Flux<T> flux;
+    private final Iterable<T> iterable;
 
     /**
-     * Creates instance with the given {@link Flux}.
+     * Creates an instance with the given {@link Flux}.
      *
      * @param flux Flux of items to iterate over.
+     * @throws NullPointerException if {@code flux} is {@code null}.
      */
     public IterableStream(Flux<T> flux) {
-        this.flux = flux;
+        this.flux = Objects.requireNonNull(flux, "'flux' cannot be null.");
+        this.iterable = null;
+    }
+
+    /**
+     * Creates an instance with the given {@link Iterable}.
+     *
+     * @param iterable Collection of items to iterate over.
+     * @throws NullPointerException if {@code iterable} is {@code null}.
+     */
+    public IterableStream(Iterable<T> iterable) {
+        this.iterable = Objects.requireNonNull(iterable, "'iterable' cannot be null.");
```

Comment:
Instead of having `flux` and `iterable` fields in this class, if an instance of IterableStream is created using an iterable, can this just be converted to `this.flux = Flux.fromIterable(Objects.requireNonNull(iterable, "'iterable' cannot be null."));`. Simplifies code in other methods too where you don't have to check if flux is null or iterable is null.

Reply:
Unfortunately this will result in that illegal state exception again because we are moving from sync to async world then back again.